tests/Unit/AbstractCompositePsrContainerTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Yiisoft\Di\Tests\Unit;
+
+use Psr\Container\ContainerInterface;
+use Yiisoft\Di\Container;
+use Yiisoft\Di\CompositeContainer;
+use Psr\Container\NotFoundExceptionInterface;
+use Yiisoft\Di\Tests\Support\EngineInterface;
+use Yiisoft\Di\Tests\Support\EngineMarkOne;
+use Yiisoft\Di\Tests\Support\EngineMarkTwo;
+
+/**
+ * General tests for PSR-11 composite container.
+ * To be extended for specific containers.
+ */
+abstract class AbstractCompositePsrContainerTest extends AbstractPsrContainerTest
+{
+    public function createCompositeContainer(?ContainerInterface $attachedContainer = null): ContainerInterface
```

Comment:
Since we're calling `attach()` and `detach()` on it, it should be more concrete.

```suggestion
    public function createCompositeContainer(?ContainerInterface $attachedContainer = null): CompositeContainer
```

Reply:
This is just a fast way to attach the first container.

Reply:
Fixed